pants-plugins/sample_conf/rules.py

```diff
@@ -51,37 +46,25 @@ class GenerateSampleConfFieldSet(FieldSet):
 
 class GenerateSampleConfViaFmtTargetsRequest(FmtTargetsRequest):
     field_set_type = GenerateSampleConfFieldSet
-    name = SCRIPT
+    tool_subsystem = ConfigGen
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
 @rule(
-    desc=f"Update conf/st2.conf.sample with {SCRIPT_DIR}/{SCRIPT}.py",
+    desc=f"Update conf/st2.conf.sample with {ConfigGen.directory}/{ConfigGen.script}.py",
     level=LogLevel.DEBUG,
 )
 async def generate_sample_conf_via_fmt(
-    request: GenerateSampleConfViaFmtTargetsRequest,
+    request: GenerateSampleConfViaFmtTargetsRequest.Batch,
+    subsystem: ConfigGen,
 ) -> FmtResult:
     # There will only be one target+field_set, but we iterate
     # to satisfy how fmt expects that there could be more than one.
     # If there is more than one, they will all get the same contents.
 
     # actually generate it with an external script.
     # Generation cannot be inlined here because it needs to import the st2 code.
```

Comment:
Does this comment make sense now we have removed this chunk of code?

Reply:
The "only one target..." bit is out of date. I dropped it.

This code block was moved, not removed. The PexFromTargetsRequest (including all of the random constants used to build it) are now defined on the subsystem. I added a comment to help clarify that.